caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -320,6 +341,28 @@ def run_tokenizer(
             TokenizationResults
                 The token count
         """
+        if self._tgis_backend:
+            self._register_model_connection_with_context(context)
+
         return self.tgis_generation_client.unary_tokenize(
             text=text,
         )
+
+    def _register_model_connection_with_context(
+        self, context: Optional[RuntimeServerContextType]
+    ):
+        """
+        Register a model connection with the configured TGISBackend.
+        """
+        ok, route_info = get_route_info(context)
+        if ok:
+            log.debug(
+                "<NLP10705560D> Registering remote model connection with context override: 'hostname: %s'",
+                route_info,
+            )
+            self._tgis_backend.register_model_connection(
+                self.base_model_name, {"hostname": route_info}, fill_with_defaults=True
+            )
+        else:
+            self._tgis_backend.register_model_connection(self.base_model_name)
```

Comment:
I think we don't need to do this if there's no context because it will be done automatically with `get_client`